app/src/androidTest/java/ch/epfl/favo/view/AddFavorTest.java

```diff
@@ -97,4 +95,222 @@ public void addFavorTest() {
+        .check(matches(isDisplayed()))
+        .check(matches(withText(fakeFavor.getStatusId().getPrettyString())));
+  }
+
+  @Test
+  public void testEditFavorFlow() {
+    Favor fakeFavor = FakeItemFactory.getFavor();
+    launchFragmentWithFakeFavor(new FavorRequestView(), fakeFavor);
+    getInstrumentation().waitForIdleSync();
+    // Click on edit
+    onView(withId(R.id.edit_favor_button))
+        .check(matches(isDisplayed()))
+        .check(matches(withText(R.string.edit_favor))) // check contains right text
+        .perform(click());
+    getInstrumentation().waitForIdleSync();
+    // Check edit button text is changed to confirm update
+    onView(withId(R.id.edit_favor_button)).check(matches(withText(R.string.confirm_favor_edit)));
+
+    // click on edit again
+    onView(withId(R.id.edit_favor_button)).perform(click());
+    getInstrumentation().waitForIdleSync();
+
+    // check snackbar shows
+    onView(withId(com.google.android.material.R.id.snackbar_text))
+        .check(matches(withText(R.string.favor_edit_success_msg)));
+  }
+
+  @Test
+  public void cancelActiveFavorUpdatesViewCorrectly() {
+    Favor fakeFavor = FakeItemFactory.getFavor();
+    launchFragmentWithFakeFavor(new FavorRequestView(), fakeFavor);
+    getInstrumentation().waitForIdleSync();
+
+    // Click on cancel
+    onView(withId(R.id.cancel_favor_button))
+        .check(matches(isDisplayed()))
+        .check(matches(withText(R.string.cancel_request))) // check contains right text
+        .perform(click());
+    getInstrumentation().waitForIdleSync();
+    // check snackbar shows
+    onView(withId(com.google.android.material.R.id.snackbar_text))
+        .check(matches(withText(R.string.favor_cancel_success_msg)));
+
+    // Check upload picture button is not clickable
+    onView(withId(R.id.add_picture_button))
+        .check(matches(isDisplayed()))
+        .check(matches(not(isEnabled())));
+
+    // Check cancel button is not clickable
+    onView(withId(R.id.cancel_favor_button)).check(matches(not(isEnabled())));
+
+    // Check updated status string
+    onView(withId(R.id.favor_status_text))
+        .check(matches(withText(Favor.Status.CANCELLED_REQUESTER.getPrettyString())));
+  }
+
+  private void launchFragmentWithFakeFavor(Fragment fragment, Favor favor) {
+    // Launch view
+    activityTestRule.getActivity().activeFavors.put(favor.getId(), favor);
+    FragmentTransaction ft =
+        activityTestRule.getActivity().getSupportFragmentManager().beginTransaction();
+    ft.replace(R.id.nav_host_fragment, FavorFragmentFactory.instantiate(favor, fragment));
+    ft.addToBackStack(null);
+    ft.commit();
+  }
+
+  public static Uri saveImageToInternalStorage(Context mContext, Bitmap bitmap) {
+
+    String mImageName = "snap.jpg";
+
+    ContextWrapper wrapper = new ContextWrapper(mContext);
+
+    File file = wrapper.getDir("Images", MODE_PRIVATE);
+
+    file = new File(file, mImageName);
+
+    try {
```

Comment:
You can use a try-with-resources here if you want, it automatically closes the stream for you and it's the recommended way to handle them ;)